data/campaigns/Delfadors_Memoirs/scenarios/12_Terror_at_the_Ford_of_Parthyn.cfg

```diff
@@ -350,10 +350,16 @@
         first_time_only=yes
         [filter]
             side=1
-            x={X}
-            y={Y}
-            radius={RADIUS}
         [/filter]
+        [filter_condition]
+            [have_location]
+                x,y=$x1,$y1
+                [and]
+                    x,y={X},{Y}
+                    radius={RADIUS}
+                [/and]
+            [/have_location]
+        [/filter_condition]
```

Comment:
OK will test. I didn't research it; just grabbed a known-working example and deleted what I didn't need.

The warning appears if the unit starts outside and lands anywhere inside, even on the ambush location. I will check that the event appears first, though, so the warning appears before the ambush (I think it does but will be sure to re-test for that). If there is a way to interrupt the move at the edge, so the user always gets the warning but is stopped from moving onto the ambush if they have not seen the warning, I'll switch to that.